src/Impostor.Server/Recorder/PacketRecorder.cs

```diff
@@ -148,11 +167,15 @@ public async Task WriteGameCreatedAsync(ClientRecorder client, GameCode gameCode
             }
         }
 
-        private static void WriteHeader(PacketSerializationContext context, RecordedPacketType type)
+        private void WriteHeader(PacketSerializationContext context, RecordedPacketType type)
         {
             // Length placeholder.
-            context.Writer.Write((int) 0);
-            context.Writer.Write((byte) type);
+            context.Writer.Write(0);
+
+            context.Writer.Write((int)(DateTimeOffset.UtcNow - _time).TotalMilliseconds);
+            _time = DateTimeOffset.UtcNow;
```

Comment:
This is for all places where you access `_time`, this is not thread-safe. This will be called by multiple threads at the same time